homeassistant/components/sensor/google_travel_time.py

```diff
@@ -78,52 +84,70 @@ def convert_time_to_utc(timestr):
-        elif hass.config.temperature_unit is TEMP_FAHRENHEIT:
-            options['units'] = 'imperial'
+        if options.get('units') is None:
+            if hass.config.temperature_unit is TEMP_CELSIUS:
+                options['units'] = 'metric'
+            elif hass.config.temperature_unit is TEMP_FAHRENHEIT:
+                options['units'] = 'imperial'
 
-    travel_mode = config.get(CONF_TRAVEL_MODE)
-    mode = options.get(CONF_MODE)
+        travel_mode = config.get(CONF_TRAVEL_MODE)
+        mode = options.get(CONF_MODE)
 
-    if travel_mode is not None:
-        wstr = ("Google Travel Time: travel_mode is deprecated, please add "
-                "mode to the options dictionary instead!")
-        _LOGGER.warning(wstr)
-        if mode is None:
-            options[CONF_MODE] = travel_mode
+        if travel_mode is not None:
+            wstr = ("Google Travel Time: travel_mode is deprecated, please "
+                    "add mode to the options dictionary instead!")
+            _LOGGER.warning(wstr)
+            if mode is None:
+                options[CONF_MODE] = travel_mode
 
-    titled_mode = options.get(CONF_MODE).title()
-    formatted_name = "Google Travel Time - {}".format(titled_mode)
-    name = config.get(CONF_NAME, formatted_name)
-    api_key = config.get(CONF_API_KEY)
-    origin = config.get(CONF_ORIGIN)
-    destination = config.get(CONF_DESTINATION)
+        titled_mode = options.get(CONF_MODE).title()
+        formatted_name = "Google Travel Time - {}".format(titled_mode)
+        name = config.get(CONF_NAME, formatted_name)
+        api_key = config.get(CONF_API_KEY)
+        origin = config.get(CONF_ORIGIN)
+        destination = config.get(CONF_DESTINATION)
 
-    sensor = GoogleTravelTimeSensor(name, api_key, origin, destination,
-                                    options)
+        sensor = GoogleTravelTimeSensor(hass, name, api_key, origin,
+                                        destination, options)
 
-    if sensor.valid_api_connection:
-        add_devices_callback([sensor])
+        if sensor.valid_api_connection:
+            add_devices_callback([sensor])
+
+    # Wait until start event is sent to load this component.
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, run_setup)
 
 
 # pylint: disable=too-many-instance-attributes
 class GoogleTravelTimeSensor(Entity):
     """Representation of a tavel time sensor."""
 
     # pylint: disable=too-many-arguments
-    def __init__(self, name, api_key, origin, destination, options):
+    def __init__(self, hass, name, api_key, origin, destination, options):
         """Initialize the sensor."""
+        self._hass = hass
         self._name = name
         self._options = options
-        self._origin = origin
-        self._destination = destination
         self._matrix = None
         self.valid_api_connection = True
 
+        # Check if location is a trackable entity
+        if any(origin.startswith(d) for d in TRACKABLE_DOMAINS):
```

Comment:
Both are O(N). Converting TRACKABLE_DOMAINS and @kellerza proposal would be O(1) 👍